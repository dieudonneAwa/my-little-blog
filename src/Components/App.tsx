import React, { useState } from "react";
import { ThemeProvider } from "styled-components";
import { ResetStyles } from "../GlobalStyles/Reset";
import { GlobalStyles } from "../GlobalStyles";
import theme from "../Theme";
import SaveUser from "./SaveUser/Form";
import ListUsers from "./ListUsers";

export type User = {
  id?: number;
  firstName: string;
  lastName: string;
  email: string;
};

function App() {
  const [users, setUsers] = useState<User[]>([])
  
  return (
    <ThemeProvider theme={theme}>
      <ResetStyles />
      <GlobalStyles />
      <SaveUser setUsers={setUsers} />
      <ListUsers users={users} />
    </ThemeProvider>
  );
}

export default App;
