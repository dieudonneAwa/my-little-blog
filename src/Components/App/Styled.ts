import styled from 'styled-components';
import { position, PositionProps, space, SpaceProps } from 'styled-system';

export const AppHeading = styled.h1<SpaceProps>`
  font-size: ${(p) => p.theme.font.size.big};
  font-weight: bold;
  margin-top: 3rem;
  margin-bottom: 3rem;
  ${space}

  @media (max-width: 414px) {
    font-size: ${(p) => p.theme.font.size.large};
    font-weight: 600;
  }
`;

export const ThemeToggler = styled.span<PositionProps>`
  cursor: pointer;
  ${position}

  &:hover {
    transition: linear 0.5s stroke;
    svg path {
      stroke: ${(p) => p.theme.colors.emerald};
    }
  }
`;
