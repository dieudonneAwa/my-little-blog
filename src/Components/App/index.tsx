import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { ResetStyles } from '../../GlobalStyles/Reset';
import { GlobalStyles } from '../../GlobalStyles';
import { theme, darkTheme } from '../../Theme';
import ArticlesPage from '../Articles';
import { Box } from '../Commons/Box';
import { Context, Provider } from '../../Context/StoreContext';
import MoonIcon from '../Commons/Icons/Moon';
import SolarIcon from '../Commons/Icons/Solar';
import { useDarkMode } from '../../Hooks/useDarkMode';
import ArticlePage from '../Article';
import { AppHeading, ThemeToggler } from './Styled';

function App() {
  const [appTheme, toggleTheme] = useDarkMode();
  const { actions, state } = useContext(Context);

  return (
    <ThemeProvider theme={appTheme === 'light' ? theme : darkTheme}>
      <ResetStyles />
      <GlobalStyles />
      <Box position="relative" display="flex" flexDirection="column" alignItems="center">
        <ThemeToggler
          position="absolute"
          right={['1rem', '1rem', "3rem"]}
          top="3rem"
          // @ts-ignore
          onClick={() => toggleTheme()}
          data-test="theme-toggler"
        >
          {appTheme === 'light' && <MoonIcon />}
          {appTheme === 'dark' && <SolarIcon fill="white" />}
        </ThemeToggler>
        <AppHeading mt={['3rem', '3rem', '3rem']}>Awa's Little Blog</AppHeading>
        <BrowserRouter>
          <Switch>
            <Route
              exact
              path="/"
              render={() => (
                <ArticlesPage
                  fetchArticles={actions.fetchArticles}
                  articles={state?.articles?.articles}
                  loading={state?.articles?.loading}
                  setArticle={actions.setArticle}
                />
              )}
            />
            <Route
              exact
              path="/:id"
              render={() => (
                <ArticlePage
                  fetchComments={actions.fetchComments}
                  comments={state.articles.comments}
                  loadingComments={state.articles.loadingComments}
                  article={state.articles.article}
                />
              )}
            />
          </Switch>
        </BrowserRouter>
      </Box>
    </ThemeProvider>
  );
}

const AppPage = () => (
  <Provider>
    <App />
  </Provider>
);

export default AppPage;
