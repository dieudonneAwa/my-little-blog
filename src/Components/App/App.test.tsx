/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { mount } from 'enzyme';
import App from './index';

describe('App', (): void => {
  let wrapper: any;
  beforeEach((): void => {
    wrapper = mount(<App />);
  });

  it('should render App component', (): void => {
    expect(wrapper).toMatchSnapshot();
  });
});
