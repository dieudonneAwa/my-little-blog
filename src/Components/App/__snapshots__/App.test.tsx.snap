// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component 1`] = `
<AppPage>
  <Provider>
    <App>
      <Component
        theme={
          Object {
            "colors": Object {
              "blueviolet": "#5621ab",
              "cultured": "#1b2546",
              "emerald": "#1EAE68",
              "lightgrey": "#11172C",
              "richblack": "#091A28",
              "spacecadet": "white",
              "white": "#1b2546",
            },
            "font": Object {
              "family": Object {
                "poppins": "'Poppins', sans-serif",
              },
              "size": Object {
                "big": "2rem",
                "huge": "3rem",
                "large": "1.5rem",
                "massive": "4rem",
                "medium": "1.1rem",
                "micro": "0.15rem",
                "mini": "0.3rem",
                "nano": "0.065rem",
                "none": "0rem",
                "small": "0.9rem",
                "tiny": "0.6rem",
              },
            },
            "radii": Object {
              "huge": "50%",
              "large": "9px",
              "medium": "6px",
              "none": "0",
              "small": "3px",
            },
            "space": Object {
              "big": "2rem",
              "huge": "3rem",
              "large": "1.5rem",
              "massive": "4rem",
              "medium": "1.1rem",
              "micro": "0.15rem",
              "mini": "0.3rem",
              "nano": "0.065rem",
              "none": "0rem",
              "small": "0.9rem",
              "tiny": "0.6rem",
            },
          }
        }
      >
        <Memo(l) />
        <Memo(l) />
        <styled.div
          alignItems="center"
          display="flex"
          flexDirection="column"
          position="relative"
        >
          <div
            className="sc-bdvvaa jfYIlR"
            display="flex"
          >
            <styled.span
              data-test="theme-toggler"
              onClick={[Function]}
              position="absolute"
              right={
                Array [
                  "1rem",
                  "1rem",
                  "3rem",
                ]
              }
              top="3rem"
            >
              <span
                className="sc-iqsfdx eGaodG"
                data-test="theme-toggler"
                onClick={[Function]}
              />
            </styled.span>
            <styled.h1
              mt={
                Array [
                  "3rem",
                  "3rem",
                  "3rem",
                ]
              }
            >
              <h1
                className="sc-kDThTU dskGWa"
              >
                Awa's Little Blog
              </h1>
            </styled.h1>
            <BrowserRouter>
              <Router
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
              >
                <Switch>
                  <Route
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    path="/"
                    render={[Function]}
                  >
                    <ArticlesPage
                      articles={Array []}
                      fetchArticles={[Function]}
                      loading={true}
                      setArticle={[Function]}
                    >
                      <styled.div
                        px="1rem"
                      >
                        <div
                          className="sc-bdvvaa hroOLu"
                        >
                          <styled.div
                            alignItems="center"
                            display="flex"
                            height="500px"
                            justifyContent="center"
                          >
                            <div
                              className="sc-bdvvaa gkiCYy"
                              display="flex"
                              height="500px"
                            >
                              <Spinner
                                color="c300"
                                size="1rem"
                              >
                                <styled.span
                                  color="c300"
                                  size="1rem"
                                >
                                  <span
                                    className="sc-gsDJrp fxttM"
                                    color="c300"
                                    size="1rem"
                                  />
                                </styled.span>
                              </Spinner>
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                    </ArticlesPage>
                  </Route>
                </Switch>
              </Router>
            </BrowserRouter>
          </div>
        </styled.div>
      </Component>
    </App>
  </Provider>
</AppPage>
`;
